docs/ruleset_functions/get_BPF_building_area_types_and_zones.md

```diff
@@ -0,0 +1,84 @@
+	MOTION_PICTURE_THEATER: ALL_OTHER,  
+	MULTIFAMILY: MULTIFAMILY,
+	MUSEUM: ALL_OTHER,  
+	OFFICE: OFFICE,  
+	PARKING_GARAGE: ALL_OTHER,  
+	PENITENTIARY: ALL_OTHER,  
+	PERFORMING_ARTS_THEATER: ALL_OTHER,  
+	POLICE_STATION: ALL_OTHER,  
+	POST_OFFICE: ALL_OTHER,  
+	RELIGIOUS_FACILITY: ALL_OTHER,  
+	RETAIL: RETAIL,  
+	SCHOOL_UNIVERSITY: SCHOOL,  
+	SPORTS_ARENA: ALL_OTHER,
+	TOWN_HALL: OFFICE,  
+	TRANSPORTATION: ALL_OTHER,  
+	WAREHOUSE: HEATED-ONLY_STORAGE,  
+	WORKSHOP: WAREHOUSE,  
+	NONE: NONE, 
+	}```
+
+- Create a dict to store results`building_area_types_with_total_area_and_zones_dict`: `building_area_types_with_total_area_and_zones_dict = {}`
+- For each building segment in RMD: `for building_segment in RMD.building.building_segments:`
+	- If the building_segment lighting_building_area_type is defined: `if building_segment.lighting_building_area_type != NULL and building_segment.lighting_building_area_type != NONE`
+		- Assign the BPF_BAT to the building_segment_BPF_BAT using the lighting_space_lookup: `building_segment_BPF_BAT = building_area_lookup[building_segment.lighting_building_area_type]`
+		- Assign "BUILDING_SEGMENT_LIGHTING" to classification_source: `classification_source = "BUILDING_SEGMENT_LIGHTING"`
+	- Else look at each zone, and then each space and determine the building_segment_BPF_BAT using the largest space.lighting_space_type: `else:`
+		- Create a dictionary to keep track of the space types and areas: `building_segment_space_types_areas_dict = {}`
+		- For each zone in the building segment: `for zone in building_segment.zones:`
+			- Create a dictionary for the zone BPF_BAT space types using the function get_zone_BPF_BAT: `zone_BPF_BAT_dict = get_zone_BPF_BAT(B_RMI, zone.id)`
+			- For each mapped building area type in the zone_BPF_BAT_dict: `for space_BPF_BAT in zone_BPF_BAT_dict:`
+				- Add this space type to the building_segment_space_types_areas_dict if it doesn't exist yet: `building_segment_space_types_areas_dict[space_BPF_BAT] = building_segment_space_types_areas_dict[space_BPF_BAT] or 0`
+				- Add the space area: `building_segment_space_types_areas_dict[space_BPF_BAT] += zone_BPF_BAT_dict[space_BPF_BAT]`
+		- Get the BPF_BAT with the largest floor area from building_segment_space_types_areas_dict: `building_segment_BPF_BAT = max(building_segment_space_types_areas_dict, key=lambda k: building_segment_space_types_areas_dict[k])`
+		- Assign "SPACE_LIGHTING" to classification_source: `classification_source = "SPACE_LIGHTING"`
+	- At this point, the building_segment_BPF_BAT has been defined by one of the three approaches, add a nested dict for this type of BPF_BAT to the building_area_types_with_total_area_and_zones_dict if it doesn't exist already: `if building_segment_BPF_BAT not in building_area_types_with_total_area_and_zones_dict: building_area_types_with_total_area_and_zones_dict[building_segment_BPF_BAT] = {"ZONE_IDS":[], "AREA":0, "CLASSIFICATION_SOURCE": classification_source}`
```

Comment:
one of the two approaches, not three.

This line will assign the first classification_source to the building_area_type.  It seems to me that it is most imporant to know the weakest classification type.  Therefore, suggest logic below to overwrite CLASSIFICATION_SOURCE if it starts as a strong source (BUILDING_SEGMENT_LIGHTING, for example) and then another portion of the building is classified with SPACE_LIGHTING.

Reply:
great catch, thanks!